src/core/src/op/squeeze.cpp

```diff
@@ -40,115 +60,69 @@ void op::Squeeze::validate_and_infer_types() {
+std::shared_ptr<Node> Squeeze::clone_with_new_inputs(const OutputVector& new_args) const {
     OV_OP_SCOPE(v0_Squeeze_clone_with_new_inputs);
     check_new_args_count(this, new_args);
-    if (new_args.size() == 1) {
-        return make_shared<Squeeze>(new_args.at(0));
-    } else if (new_args.size() == 2) {
-        return make_shared<Squeeze>(new_args.at(0), new_args.at(1));
-    } else {
+
+    switch (new_args.size()) {
+    case 1:
+        return std::make_shared<Squeeze>(new_args[0]);
+    case 2:
+        return std::make_shared<Squeeze>(new_args[0], new_args[1]);
+    default:
         OPENVINO_THROW("Incorrect number of new arguments");
     }
 }
 
-OPENVINO_SUPPRESS_DEPRECATED_START
-bool op::v0::Squeeze::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+bool Squeeze::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v0_Squeeze_evaluate);
-    OPENVINO_SUPPRESS_DEPRECATED_START
-    OPENVINO_ASSERT(validate_host_tensor_vector(inputs, inputs.size()));
-    OPENVINO_ASSERT(validate_host_tensor_vector(outputs, 1));
-    OPENVINO_SUPPRESS_DEPRECATED_END
-
-    if (has_evaluate()) {
-        auto input_shapes = std::vector<PartialShape>{inputs[0]->get_partial_shape()};
-
-        if (inputs.size() == 2) {
-            input_shapes.push_back(inputs[1]->get_partial_shape());
-        }
-
-        auto output_shapes = shape_infer(this, input_shapes, make_tensor_accessor(inputs));
+    OPENVINO_ASSERT(outputs.size() == 1);
 
-        auto out_shape = output_shapes[0].get_shape();
-        outputs[0]->set_shape(out_shape);
+    const auto output_shapes = shape_infer(this, util::get_tensors_shapes(inputs), make_tensor_accessor(inputs));
+    outputs[0].set_shape(output_shapes.front().get_shape());
 
-        ov::reference::copy(inputs[0]->get_data_ptr<char>(),
-                            outputs[0]->get_data_ptr<char>(),
-                            shape_size(out_shape) * outputs[0]->get_element_type().size());
-
-        return true;
-    }
-    return false;
+    std::memcpy(outputs[0].data(), inputs[0].data(), outputs[0].get_byte_size());
+    return true;
 }
-OPENVINO_SUPPRESS_DEPRECATED_END
 
-bool op::v0::Squeeze::has_evaluate() const {
+bool Squeeze::has_evaluate() const {
     OV_OP_SCOPE(v0_Squeeze_has_evaluate);
-
-    if (get_input_size() == 2) {
-        switch (get_input_element_type(1)) {
-        case ngraph::element::i8:
-        case ngraph::element::i16:
-        case ngraph::element::i32:
-        case ngraph::element::i64:
-        case ngraph::element::u8:
-        case ngraph::element::u16:
-        case ngraph::element::u32:
-        case ngraph::element::u64:
-            return true;
-        default:
-            break;
-        }
-        return false;
-    } else if (get_input_size() == 1) {
-        return true;
-    } else {
-        return false;
-    }
+    return (get_input_size() < 2) || validate::axes_type(get_input_element_type(1));
```

Comment:
nit: Maybe we can have internal lambda instead of introducing ` validate::axes_type`